### ISSUE TITLE
proposal: optional chaining is conditionality and should violate `no-conditional-in-test`

### ISSUE THREAD

**Issue by jroru** [vitest-dev/eslint-plugin-vitest#643]
This code currently violates `no-conditional-in-test`
```ts
describe("suite", () => {
  it("clicking button does something", () => {
    const button: Button | null = findButton();
    if (button) {
      button.click();
    }
    expect(someCallback).toHaveBeenCalled();
  });
});
```

Whereas this code does not:
```ts
describe("suite", () => {
  it("clicking button does something", () => {
    const button: Button | null = findButton();
    button?.click();
    expect(someCallback).toHaveBeenCalled();
  });
});
```

Happy to help on a PR if there is consensus.

**Comment by veritem** [comment]
i agree, a pr is welcome!
